apps/api/v2/src/modules/auth/guards/or-auth/or-auth.guard.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable, CanActivate, ExecutionContext, Type, Logger } from "@nestjs/common";
+import { ModuleRef } from "@nestjs/core";
+
+/**
+ * Decorator function that creates an OrAuth guard with the specified guards
+ * @param guards Array of guard classes to evaluate with OR logic
+ * @returns A guard class that grants access if ANY of the provided guards return true
+ */
+export function OrAuth(guards: Type<CanActivate>[]) {
```

Comment:
I would rename it `OrGuard` because `OrAuth` sounds like it accepts auth methods like api key, access token (which we already have FYI)
```
@UseGuards(ApiAuthGuard)
@ApiAuthGuardOnlyAllow(["API_KEY", "ACCESS_TOKEN"])
```

This guard is used as `OrAuth([RolesGuard, IsUserRoutingForm]` and none of the guards are auth related, so `OrGuard` is a better name or even `OneOf` aka
```
@UseGuards(@OneOf([RolesGuard, IsUserRoutingForm]))
```

Reply:
Oh, I just used what Morgan shared as an example. Sorry I will fix it.

_OneOf_ sounds good

Reply:
Using **_Or_**